NanaZip.UI.Classic/SevenZip/CPP/7zip/UI/FileManager/PanelFolderChange.cpp

```diff
@@ -75,7 +75,26 @@ static bool DoesNameContainWildcard_SkipRoot(const UString &path)
 
 HRESULT CPanel::BindToPath(const UString &fullPath, const UString &arcFormat, COpenResult &openRes)
 {
-  UString path = fullPath;
+  
+  // **************** NanaZip Modification Start **************
+  //UString path = fullPath;
+  wchar_t Buffer[MAX_PATH * 4] = {};
```

Comment:
What's meaning of MAX_PATH * 4? I think 32767 will be better because it's the NT and Win32 UNC path's maximum length limit.

I think you can get some improvement hints from the following implementation, which is referenced in https://github.com/M2Team/NanaBox/blob/main/NanaBox/Utils.cpp

```cpp
std::wstring GetCurrentProcessModulePath()
{
    // 32767 is the maximum path length without the terminating null character.
    std::wstring Path(32767, L'\0');
    Path.resize(::GetModuleFileNameW(
        nullptr, &Path[0], static_cast<DWORD>(Path.size())));
    return Path;
}
```

Reply:
> You'd need to use 32768 due to null-terminator.

For wchar_t scenario, it need 32768. For the std::string and std::wstring scenario, it actually only need 32767 because null terminator will be included in that.